vars/artifactSetVersion.groovy

```diff
@@ -0,0 +1,138 @@
+import com.sap.piper.ConfigurationLoader
+import com.sap.piper.ConfigurationMerger
+import com.sap.piper.GitUtils
+import com.sap.piper.Utils
+import com.sap.piper.versioning.ArtifactVersioning
+
+import groovy.text.SimpleTemplateEngine
+
+def call(Map parameters = [:]) {
+
+    def stepName = 'artifactSetVersion'
+
+    handlePipelineStepErrors (stepName: stepName, stepParameters: parameters) {
+
+        def gitUtils = parameters.juStabGitUtils
+        if (gitUtils == null) {
+            gitUtils = new GitUtils()
+        }
+
+        //ToDo: exit if pending git changes are in the current repo? TBD!
+
+        def script = parameters.script
+        if (script == null)
+            script = [commonPipelineEnvironment: commonPipelineEnvironment]
+
+        prepareDefaultValues script: script
+
+        final Map stepDefaults = ConfigurationLoader.defaultStepConfiguration(script, stepName)
+        final Map stepConfiguration = ConfigurationLoader.stepConfiguration(script, stepName)
+
+        List parameterKeys = [
+            'artifactType',
+            'buildTool',
+            'dockerVersionSource',
+            'filePath',
+            'gitCommitId',
+            'gitCredentialsId',
+            'gitUserEMail',
+            'gitUserName',
+            'gitSshUrl',
+            'tagPrefix',
+            'timestamp',
+            'timestampTemplate',
+            'versioningTemplate'
+        ]
+        Map pipelineDataMap = [
+            gitCommitId: gitUtils.getGitCommitId()
+        ]
+        List stepConfigurationKeys = [
+            'artifactType',
+            'buildTool',
+            'dockerVersionSource',
+            'filePath',
+            'gitCredentialsId',
+            'gitUserEMail',
+            'gitUserName',
+            'gitSshUrl',
+            'tagPrefix',
+            'timestamp',
+            'timestampTemplate',
+            'versioningTemplate'
+        ]
+
+        Map configuration = ConfigurationMerger.mergeWithPipelineData(parameters, parameterKeys, pipelineDataMap, stepConfiguration, stepConfigurationKeys, stepDefaults)
+
+        //buildTool is mandatory parameter
+        def utils = new Utils()
+        def buildTool = utils.getMandatoryParameter(configuration, 'buildTool')
+
+        if (!configuration.filePath)
+            configuration.filePath = configuration[buildTool].filePath //use default configuration
+
+        def newVersion
+        def artifactVersioning = ArtifactVersioning.getArtifactVersioning(buildTool, this, configuration)
+
+        if(configuration.artifactType == 'appContainer' && configuration.dockerVersionSource == 'appVersion'){
```

Comment:
This is needed here to cater for a very specific case:
Having first an artifact build and then a container build in the same pipeline script - where the container gets the version of the previously built artifact (kind of Docker „builder“ pattern)

We should not get any other conditionals here since all other cases should be handled with the pattern based versioning.